mesonbuild/wrap/wrap.py

```diff
@@ -372,12 +372,16 @@ def resolve_git_submodule(self) -> bool:
         # definitely cannot try to conveniently set up a submodule.
         if not GIT:
             return False
+        # Does the directory exist? Even uninitialised submodules checkout an
+        # empty directory to work in
+        if not os.path.isdir(self.dirname):
+            return False
         # Are we in a git repository?
-        ret, out = quiet_git(['rev-parse'], self.subdir_root)
+        ret, out = quiet_git(['rev-parse'], str(Path(self.dirname).parent))
```

Comment:
Hmm, I just realized. You are converting a Path to a str, because mesonlib.quiet_git requires a str there... but it is directly passed to subprocess.Popen as `cwd=`, and that accepts os.PathLike.

So, if we just fix that typing issue...

Reply:
#9888

Reply:
There appears to be a lot of code calling the `*git` functions - is mypy happy with passing a string to a signature accepting a PathLike? Otherwise it's a much bigger change.

Also is the expectation that I make the change? I'm more than happy to, just don't want to duplicate work if you're planning on it :)

Reply:
My PR declares support for anything that subprocess.Popen itself takes, i.e. `T.Union[str, bytes, os.PathLike]`, so mypy should be happy. :)

Reply:
TBH, in the long run we kind of want to use fewer raw strings anyway, PathLike objects like pathlib.Path are really nice for this exact case, so... fixing `*git()` is just a good idea no matter what.

Reply:
Oh lovely, of course just making them take a union works.

I'll wait for that to merge, then rebase again without the cast!

Reply:
Merged, please rebase. :)